File: Treachery-iOS/Treachery-iOS/Models/GameMode.swift
//
//  GameMode.swift
//  Treachery-iOS
//
//  Created by Luke Solomon on 3/16/26.
//

import Foundation

enum GameMode: String, Codable, CaseIterable {
    case treachery
    case planechase
    case treacheryPlanechase = "treachery_planechase"
    case none

    var displayName: String {
        switch self {
        case .treachery: return "Treachery"
        case .planechase: return "Planechase"
        case .treacheryPlanechase: return "Both"
        case .none: return "Life Tracker"
        }
    }

    var includesTreachery: Bool {
        self == .treachery || self == .treacheryPlanechase
    }

    var includesPlanechase: Bool {
        self == .planechase || self == .treacheryPlanechase
    }
}

File: TreacheryAndroid/app/src/main/java/com/solomon/treachery/model/GameMode.kt
package com.solomon.treachery.model

enum class GameMode(val value: String) {
    TREACHERY("treachery"),
    PLANECHASE("planechase"),
    TREACHERY_PLANECHASE("treachery_planechase"),
    NONE("none");

    val displayName: String
        get() = when (this) {
            TREACHERY -> "Treachery"
            PLANECHASE -> "Planechase"
            TREACHERY_PLANECHASE -> "Both"
            NONE -> "Life Tracker"
        }

    val includesTreachery: Boolean
        get() = this == TREACHERY || this == TREACHERY_PLANECHASE

    val includesPlanechase: Boolean
        get() = this == PLANECHASE || this == TREACHERY_PLANECHASE

    companion object {
        fun fromValue(value: String): GameMode? = entries.find { it.value == value }
    }
}
